From b72e9e3b728daa958c968bc084c5ed74778fcb56 Mon Sep 17 00:00:00 2001
From: nx111 <gd.zhangdz@gmail.com>
Date: Fri, 26 Jan 2018 09:01:22 +0800
Subject: [PATCH 2/8] Revert "SimSettings: Correctly disable sim switch without
 SIM card"

This reverts commit ab2e226fc95473121eaccc673b21ab56bba90f4c.
---
 src/com/android/settings/sim/SimSettings.java | 24 ++++--------------------
 1 file changed, 4 insertions(+), 20 deletions(-)

diff --git a/src/com/android/settings/sim/SimSettings.java b/src/com/android/settings/sim/SimSettings.java
index 2d71c0c..07ea2c3 100644
--- a/src/com/android/settings/sim/SimSettings.java
+++ b/src/com/android/settings/sim/SimSettings.java
@@ -476,11 +476,7 @@ public class SimSettings extends RestrictedSettingsFragment implements Indexable
                 mSwitch.setVisibility(View.GONE);
             } else {
                 mSwitch.setVisibility(View.VISIBLE);
-                if (isAirplaneModeOn() || (!isCurrentSubValid())) {
-                    mSwitch.setEnabled(false);
-                } else {
-                    mSwitch.setEnabled(true);
-                }
+                mSwitch.setEnabled(!isAirplaneModeOn() || isValid());
                 setChecked(getProvisionStatus(mSlotId) == PROVISIONED);
             }
         }
@@ -509,21 +505,9 @@ public class SimSettings extends RestrictedSettingsFragment implements Indexable
 
         // This method returns true if SubScription record corresponds to this
         // Preference screen has a valid SIM and slot index/SubId.
-        private boolean isCurrentSubValid() {
-            boolean isSubValid = false;
-            if (hasCard()) {
-                SubscriptionInfo sir = mSubscriptionManager.
-                        getActiveSubscriptionInfoForSimSlotIndex(mSlotId);
-                if (sir != null ) {
-                    mSubInfoRecord = sir;
-                    if (SubscriptionManager.isValidSubscriptionId(mSubInfoRecord.getSubscriptionId()) &&
-                            mSubInfoRecord.getSimSlotIndex() >= 0 &&
-                            getProvisionStatus(mSubInfoRecord.getSimSlotIndex()) >= 0) {
-                        isSubValid = true;
-                    }
-                }
-            }
-            return isSubValid;
+        @Override
+        protected boolean isValid() {
+            return super.isValid() && getProvisionStatus(mSlotId) >= 0;
         }
 
         // Based on the received SIM provision state this method
-- 
2.7.4

